src/test/scala/is/hail/expr/ir/OrderingSuite.scala

```diff
@@ -1,17 +1,46 @@
 package is.hail.expr.ir
 
-import is.hail.annotations.{CodeOrdering, Region, RegionValueBuilder}
+import is.hail.annotations.{Region, RegionValueBuilder, SafeIndexedSeq}
 import is.hail.check.{Gen, Prop}
 import is.hail.asm4s._
 import is.hail.expr.ir._
 import is.hail.expr.types._
-import is.hail.utils.Interval
+import is.hail.utils._
 import org.apache.spark.sql.Row
 import org.testng.annotations.Test
 
 class OrderingSuite {
 
-  @Test def testRandomAgainstUnsafe() {
+  def recursiveSize(t: Type): Int = {
+    val inner = t match {
+      case ti: TInterval => recursiveSize(ti.pointType)
+      case tc: TContainer => recursiveSize(tc.elementType)
+      case tbs: TBaseStruct =>
+        tbs.types.map{ t => recursiveSize(t) }.sum
+      case _ => 0
+    }
+    inner + 1
+  }
+
+  def getStagedOrderingFunction[T: TypeInfo](t: Type, comp: String): AsmFunction3[Region, Long, Long, T] = {
+    val fb = EmitFunctionBuilder[Region, Long, Long, T]
+    val stagedOrdering = t.codeOrdering(fb.apply_method)
+    val cregion: Code[Region] = fb.getArg[Region](1)
+    val cv1 = coerce[stagedOrdering.T](cregion.getIRIntermediate(t)(fb.getArg[Long](2)))
+    val cv2 = coerce[stagedOrdering.T](cregion.getIRIntermediate(t)(fb.getArg[Long](3)))
+    comp match {
+      case "compare" => fb.emit(stagedOrdering.compare(cregion, (const(false), cv1), cregion, (const(false), cv2)))
+      case "equiv" => fb.emit(stagedOrdering.equiv(cregion, (const(false), cv1), cregion, (const(false), cv2)))
+      case "lt" => fb.emit(stagedOrdering.lt(cregion, (const(false), cv1), cregion, (const(false), cv2)))
+      case "lteq" => fb.emit(stagedOrdering.lteq(cregion, (const(false), cv1), cregion, (const(false), cv2)))
+      case "gt" => fb.emit(stagedOrdering.gt(cregion, (const(false), cv1), cregion, (const(false), cv2)))
+      case "gteq" => fb.emit(stagedOrdering.gteq(cregion, (const(false), cv1), cregion, (const(false), cv2)))
+    }
+
+    fb.result()()
+  }
+
+  @Test def testRandomOpsAgainstExtended() {
```

Comment:
The random tests only do 10 by default. Can you crank this up to a big number and let it run a few minutes locally?

Reply:
yeah, I did size = 10 and count = 1000 before I PR'ed. I could crank up both and try again, if you want.

Reply:
that's plenty for me